src/Microsoft.Sbom.Api/Workflows/Helpers/RelationshipsArrayGenerator.cs

```diff
@@ -26,92 +26,87 @@ public class RelationshipsArrayGenerator : IJsonArrayGenerator<RelationshipsArra
 
     private readonly ILogger log;
 
-    private readonly ISbomConfigProvider sbomConfigs;
-
     private readonly IRecorder recorder;
 
+    private GenerationData generationData;
+
+    public ISbomConfig SbomConfig { get; set; }
+
+    public string SpdxManifestVersion { get; set; }
+
     public RelationshipsArrayGenerator(
         RelationshipGenerator generator,
         ChannelUtils channelUtils,
         ILogger log,
-        ISbomConfigProvider sbomConfigs,
         IRecorder recorder)
     {
         this.generator = generator;
         this.channelUtils = channelUtils;
         this.log = log;
-        this.sbomConfigs = sbomConfigs;
         this.recorder = recorder;
+        this.generationData = this.SbomConfig.Recorder.GetGenerationData();
```

Comment:
At construction, `SbomConfig` is null, so this is throwing a `NullReferenceException`. If the intent is to call `SbomConfig.Recorder.GetGenerationData` only once, you need to defer the call until after `SbomConfig` has been set. You could initialize it in the setter for `SbomConfig`, or you could use a `Lazy` to defer reading it from `SbomConfig` until `generationData` is actually read. Happy to discuss this directly, if you wish.

Reply:
I had initially moved it to the constructor since we didn't want to have `this.` outside of the constructor, but I don't think there's a way around that here since sbomConfig needs to be set before calling `GetGenerationData()`